src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -16,20 +16,20 @@
 namespace Elastic.Apm.AspNetCore
 {
 	// ReSharper disable once ClassNeverInstantiated.Global
-	public class ApmMiddleware
+	internal class ApmMiddleware
 	{
 		private readonly RequestDelegate _next;
-		private readonly ITracer _tracer;
+		private readonly Tracer _tracer;
 
-		public ApmMiddleware(RequestDelegate next, ITracer tracer)
+		public ApmMiddleware(RequestDelegate next, Tracer tracer)
 		{
 			_next = next;
 			_tracer = tracer;
 		}
 
 		public async Task InvokeAsync(HttpContext context)
 		{
-			var transaction = _tracer.StartTransaction($"{context.Request.Method} {context.Request.Path}",
+			var transaction = _tracer.StartTransactionInternal($"{context.Request.Method} {context.Request.Path}",
 				ApiConstants.TypeRequest);

```

Comment:
Not related to the changeset but more with initial exposore and naming/usage of `ApiConstants`. 

Instead of `type` being `string` we can have it accept `TransactionType`:

```
public struct TransactionType 
{
    public static readonly TransactionType Request = new TransactionType("request");

    private readonly string _name;
    public TransactionType(string name) => _name = name;
    
    public static implicit operator TransactionType(string name) => new TransactionType(name);

     public override string ToString() => _name;
}
```

If the last param to `StartTransaction` then takes `TransactionType` both of these will be valid:

```csharp
_.StartTransaction("a name", TransactionType.Request);
```

```csharp
_.StartTransaction("a name", "any type");
```

This would break the types we ship with down more clearly between `TransactionType` `SpanType` and `ActionType`.

Reply:
👍 That sounds good! There are also `action` and `subtype` on span (https://github.com/elastic/apm/issues/8). We should also encapsulate those.